File: 260321/그 요일은/the-day-of-the-day.py
m1, d1, m2, d2 = map(int, input().split())
A = input()

month_days = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
day_weekend = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']

def num_days(month, day):
    total = 0
    for m in range(1, month):
        total += month_days[m]
    return total + day - 1

diff = num_days(m2, d2) - num_days(m1, d1)
num_week = diff // 7
weekend = day_weekend[diff % 7]
count = 0


count += num_week
if weekend == A:
    count += 1

print(count)

File: code_tree/trail2/그 요일은/the-day-of-the-day.cpp
#include <iostream>

using namespace std;

int m1, m2, d1, d2;
string A;

int DayNums(int m1, int d1, int m2, int d2) {
    int days[13] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int total_days = 0;
    for (int i = m1; i < m2; i++) 
        total_days += days[i];
    return total_days - d1 + d2;
}

int Day(string a) {
    if (a == "Mon") return 0;
    else if (a == "Tue") return 1;
    else if (a == "Wed") return 2;
    else if (a == "Thu") return 3;
    else if (a == "Fri") return 4;
    else if (a == "Sat") return 5;
    else if (a == "Sun") return 6;
}

int main() {
    cin >> m1 >> d1 >> m2 >> d2;
    cin >> A;

    int diff = DayNums(m1, d1, m2, d2);

    int num = Day(A);
    int cnt = 0;


    for (int i = 0; i <= diff; i++) {
        if ((i % 7) == num)
            cnt++;
    }

    cout << cnt << endl;


    return 0;
}